### PR TITLE
[WIP] Add Face Variables To Restart & Graphics Dumps

### PR DESCRIPTION
**Pull request by nmsriram**
## PR Summary
Provides the capability to write face based variables to graphics and restart dumps.  Currently the only way to retrieve face based data from graphics dumps is to use hdf tools, but we hope to identify a canonical means of visualizing this output in upcoming commits.

Note that although we write variables, they are not currently read back in.  One of the tests will be modified to read the face variables in the unit tests.

Addresses Issue #360 
## PR Checklist

- [x] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [ ] Changes are summarized in CHANGELOG.md

### CODE REVIEW

File: src/bvals/boundary_conditions.cpp [hunk 1]
```diff
@@ -36,12 +36,12 @@ TaskStatus ApplyBoundaryConditions(std::shared_ptr<MeshBlockData<Real>> &rc) {
...
   Metadata m;
   MeshBlockDataIterator<Real> citer(rc, {Metadata::Independent});
-  const int nvars = citer.vars.size();
+  const int nvars = citer.varsCell.size();
 
   switch (pmb->boundary_flag[BoundaryFace::inner_x1]) {
   case BoundaryFlag::outflow:
     for (int n = 0; n < nvars; n++) {
-      ParArrayND<Real> q = citer.vars[n]->data;
+      ParArrayND<Real> q = citer.varsCell[n]->data;
```
Comment (Yurlungur) [inline src/bvals/boundary_conditions.cpp:44]: This is going to change substantially when #332 goes in.

File: src/outputs/restart.cpp [hunk 2]
```diff
@@ -469,6 +469,111 @@ void RestartOutput::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm)
...
+    }
+    // close data spaces
+    H5Sclose(vLocalSpaceX);
+    H5Sclose(vGlobalSpaceX);
+    H5Sclose(vLocalSpaceY);
+    H5Sclose(vGlobalSpaceY);
+    H5Sclose(vLocalSpaceZ);
+    H5Sclose(vGlobalSpaceZ);
+  }
+
```
Comment (Yurlungur) [inline src/outputs/restart.cpp:576]: This is almost line-for-line duplicated between restart and output. Can it be abstracted a bit. Even if it has to eventually change, it'd be better to change things in only one spot.

File: src/outputs/parthenon_hdf5.cpp [hunk 3]
```diff
@@ -445,6 +445,113 @@ void PHDF5Output::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm) {
...
+      auto ci =
+          MeshBlockDataIterator<Real>(pmb->meshblock_data.Get(), output_params.variables);
+      for (auto &v : ci.varsFace) {
+        std::string name = v->label();
+        if (name.compare(vWriteName) == 0) {
+          // copy data to host
+
+          // Load x direction
+          auto v_x = v->Get(1).GetHostMirrorAndCopy();
+          LOADVARIABLEONE(index1, dataF, v_x, out_ib.s, out_ib.e + 1, out_jb.s, out_jb.e,
```
Comment (Yurlungur) [inline src/outputs/parthenon_hdf5.cpp:503]: I take it these macros are already implemented in parthenon?

**Comment by jlippuner** [comment]
Probably deprecated by new IO stuff

**Pull request closed without merging**